chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -78,12 +80,31 @@ impl Client {
             chunk.chunk_hash(),
             chunk_validators,
         );
+
+        // Record the witness in order to match the incoming acks for measuring round-trip times.
+        // See process_chunk_state_witness_ack for the handling of the ack messages.
+        self.state_witness_tracker.record_witness_sent(&witness, witness_size,
+                                                       chunk_validators.len());
+
         self.network_adapter.send(PeerManagerMessageRequest::NetworkRequests(
             NetworkRequests::ChunkStateWitness(chunk_validators, witness),
         ));
         Ok(())
     }
 
+    /// Handles the state witness ack message from the chunk validator.
+    /// It computes the round-trip time between sending the state witness and receiving
+    /// the ack message and updates the corresponding metric with it.
+    pub fn process_chunk_state_witness_ack(
+        &mut self,
+        witness_ack: ChunkStateWitnessAck
+    ) -> Result<(), Error> {
+        self.state_witness_tracker.on_witness_ack_received(witness_ack);
+        // NOTE: Currently we do not raise an error for ack message handling,
```

Comment:
if we don't raise any errors then why do we want to return `Result` here? maybe just remove return type completely

Reply:
the caller of process_chunk_state_witness_ack (ClientActions) is implementing ClientActionHandler<ChunkStateWitnessAckMessage> and the return type comes from the trait definition. but you are right that this function does not need to return Result, moved the Ok return to the caller side.